Proteomics/ProteolyticDigestion/Protease.cs

```diff
@@ -172,13 +172,28 @@ internal List<int> GetDigestionSiteIndices(string proteinSequence)
             var indices = new List<int>();
             for (int r = 0; r < proteinSequence.Length; r++)
             {
+                var indice = -1;
+                var preventing = new List<DigestionMotif>();
+
                 foreach (DigestionMotif motif in DigestionMotifs)
                 {
-                    if (motif.Fits(proteinSequence, r) && r + motif.CutIndex < proteinSequence.Length)
+                    var fits = motif.Fits(proteinSequence, r);
+                    if (fits.Item1 && r + motif.CutIndex < proteinSequence.Length)
+                    {
+                        indice = Math.Max(r + motif.CutIndex, indice);
+                    }
+
+                    if (fits.Item2) // if any motif prevents cleave
                     {
-                        indices.Add(r + motif.CutIndex);
+                        preventing.Add(motif);
```

Comment:
I think I see what you're doing here - you're saying that if one motif is prevented, then all motifs are prevented from cleaving at this site. I don't think we want to do that - can you revert all changes to this file?